### PR TITLE
doc: new release cycle

### PR DESCRIPTION
**Pull request by romani**
latest:
image

without "when maintainers wants"
image

all
image

### CODE REVIEW

**Comment by rnveach** [comment]
I disagree with releasing so often. I can see us doing it on some requests, but I think we should stick to the monthly releases to get more into the release instead of 1 fix.

**Comment by romani** [comment]
We will release only one fix if author of merged Pull request need it quicker. I doubt it will happen often.
It is good to release before breaking changes to let authors of previous fixes not be affected.

I very confident that we will stay 1 month, with 1-2 more mid month release per year.

**Comment by romani** [comment]
We will release only one fix if author of merged Pull request need it quicker. I doubt it will happen often.
It is good to release before breaking changes to let authors of previous fixes not be affected.

I very confident that we will stay 1 month, with 1-2 more mid month release per year.

**Comment by romani** [comment]
@nrmancuso , CI should be green, please share feedback.

**Comment by romani** [comment]
By automation of release we resolved big problem of open source - after contribution, authors need to wait long time to let fixes be released.  The longer period of wait, the bigger discourage to contribute. In most cases it is easier to make hack 😜 in target environment and never bother to contribute.

Imagine defect is found in some proprietary code base, if engineer knows that it will be released same day , he will be more likely to encourage him company to do a fix in open source project, rather than hacking something and doing workarounds.

With delayed release, workaround is not avoidable, not much insensitive to contribute.

It same as purchase of something online with same day delivery, it is awesome.

**Comment by rnveach** [comment]
>  The longer period of wait, the bigger discourage to contribute.

I think we have other, probably bigger issues discouraging contribution besides waiting 30 days or less for a release.

**Comment by romani** [comment]
Github, generate site

**Comment by nrmancuso** [comment]
This seems like a lot of words to say "we will release whenever we want" ;)

I am not sure about the value of this section to users. Maybe we need to revisit the entire release concept if we need a document with a list of conditions that pertain to release.

It might be better to have a nightly maven deployment and completely automated monthly release (no human involvement in the release process at all).

**Comment by romani** [comment]
>This seems like a lot of words to say "we will release whenever we want" ;)

No, at end of month, or by special request. 
We did this even before this automation.

>I am not sure about the value of this section to users. Maybe we need to revisit the entire release concept if we need a document with a list of conditions that pertain to release.

We can remove all except note to contributors that they can request ( not a guarantee to be done immediately).
I just did extended description as I got a lot of questions about this.

>It might be better to have a nightly maven deployment and completely automated monthly release

It is far future. I still think that daily releases are a bit noisy, we do have regression leak sometimes to master.  We can think about this in future, not now.

**Comment by romani** [comment]
@rnveach , are you ok to keep only one note ?

Release before breaking changes merge is also good, to respect all contributors that merged their stuff before but didn't not required quick release.

**Comment by nrmancuso** [comment]
>Release before breaking changes merge is also good, to respect all contributors that merged their stuff before but didn't not required quick release.

Concept is OK, contract in documentation to honor it is probably going to get forgotten sometimes.

**Comment by romani** [comment]
Everything can be forgotten and ignored and being blocked by technical problems.
All we do is our best effort, not a guarantee.

**Comment by rnveach** [comment]
> We can remove all except note to contributors that they can request ( not a guarantee to be done immediately).

That is fine. Old wording was "will happen", there was no wiggle room or anything. I am more for keeping once a month releases than 1 commit releases.

**Comment by romani** [comment]
@rnveach , @nrmancuso , 

>before merging to master branch any breaking compatibility changes

please confirm that you want this to be removed. This is good practice, I did not make it before only because release process was expensive.

**Comment by romani** [comment]
Github, generate site

**Comment by rnveach** [comment]
````
      Releases happen at the end of each month.
       Releases can happen by request of author of merged Pull Request(s), but it is not always guaranteed.

````

**Comment by romani** [comment]
done.

**Review by rnveach — APPROVED** [review]

**Comment by romani** [comment]
Github, generate site

File: src/xdocs/index.xml.vm
Comment (nrmancuso) [inline src/xdocs/index.xml.vm:97]: This seems unnecessarily restrictive. What if I report a bug, but someone else fixes it?  Can I request a release?

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Question:

File: src/xdocs/index.xml.vm
Comment (romani) [reply src/xdocs/index.xml.vm:97]: I am ok to release by any request, as it is single click, we can do this from phones at any time. If someone needs a fix to be released, let's do it.
I doubt it would happen frequently, as usually who do fix needs it, all others might not exposed to such problem.

Comment (romani) [reply src/xdocs/index.xml.vm:97]: @rnveach , do you agree ?

`can happen by request of author of merged Pull any user who is affected and what to use fix quicker`

Comment (rnveach) [reply src/xdocs/index.xml.vm:97]:
````
can happen by request by any user who is impacted, but it is not always guaranteed.
````

Comment (romani) [reply src/xdocs/index.xml.vm:97]: done.

**Comment by romani** [comment]
Github, generate site

**Review by nrmancuso — APPROVED** [review]

**Pull request merged**
